passthrough-server/src/main/java/org/terracotta/passthrough/PassthroughConnection.java

```diff
@@ -318,34 +316,43 @@ public Void decode(Type type, boolean shouldReplicate, long transactionID, long
   }
 
   private void handleAck(PassthroughServerProcess sender, long transactionID) {
-    PassthroughWait waiter = this.connectionState.getWaiterForTransaction(sender, transactionID);
+    PassthroughInvocationCallback invocation = this.connectionState.getInvocationForTransaction(sender, transactionID);
     // Note that we may fail because this server may be dead.
-    if (null != waiter) {
-      waiter.handleAck();
+    if (null != invocation) {
+      invocation.received();
     }
   }
 
   private void handleComplete(PassthroughServerProcess sender, long transactionID, byte[] result, EntityException error) {
-    PassthroughWait waiter = this.connectionState.getWaiterForTransaction(sender, transactionID);
+    PassthroughInvocationCallback invocation = this.connectionState.getInvocationForTransaction(sender, transactionID);
     // Note that we may fail because this server may be dead.
-    if (null != waiter) {
-      waiter.handleComplete(result, error);
+    if (null != invocation) {
+      try {
```

Comment:
Not going to argue about this but, since the completion state of a message is either some response object or an exception, it's kind of wierd to accept both here ... but I get it.